### PR TITLE
IGVF-202 Added organism derived from biosample.

### PR DESCRIPTION
**Pull request by KAndreeva**
Added new organism schema derived from biosample.
Added change log for organism schema.
Added schema tests for organism.
Added inserts for organism.

### CODE REVIEW

File: src/igvfd/schemas/whole_organism.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        "sex": {
+            "title": "Sex",
+            "description": "Sex of the whole organism sample.",
+            "type": "string",
+            "default": "unspecified",
+            "enum": [
+                "male",
+                "female",
+                "unspecified"
+            ]
```
Comment (gabdank) [inline src/igvfd/schemas/whole_organism.json:63]: I think this is a calculated property

File: src/igvfd/tests/data/inserts/whole_organism.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        "description": "frozen yeast sample",
+        "taxa": "Saccharomyces"
+    },
+    {
+        "uuid": "2380761b-a127-4af8-92e7-63cc53828308",
+        "accession": "IGVFSM002KAS",
+        "lab": "j-michael-cherry",
+        "award": "HG012012",
+        "aliases": [
+            "j-michael-cherry:E_colli"
```
Comment (gabdank) [inline src/igvfd/tests/data/inserts/whole_organism.json:40]: I think you misspelled e.coli (not a double l)

Comment (gabdank) [reply src/igvfd/tests/data/inserts/whole_organism.json:40]: probably needs taxa as well?

File: src/igvfd/tests/test_schema_whole_organism.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    res = testapp.patch_json(
+        whole_organism1['@id'],
+        {'description': 'I am a modern major general.'})
+    assert res.status_code == 200
+
+
+def test_failure_patch_calculated_sex(whole_organism1, testapp):
+    res = testapp.patch_json(
+        whole_organism1['@id'],
+        {'sex': 'female'}, expect_errors=True)
```
Comment (gabdank) [inline src/igvfd/tests/test_schema_whole_organism.py:26]: you are testing that the sex is calculated, but haven't removed it from the schema?

File: src/igvfd/schemas/whole_organism.json (hunk 1 above)
Comment (gabdank) [reply src/igvfd/schemas/whole_organism.json:63]: i think should be removed form the schema

**Review by gabdank — CHANGES_REQUESTED** [review]

File: src/igvfd/schemas/whole_organism.json (hunk 1 above)
Comment (KAndreeva) [reply src/igvfd/schemas/whole_organism.json:63]: I saw that and I changed my test, but I forgot to remove it from here:( Done!

File: src/igvfd/tests/test_schema_whole_organism.py (hunk 3 above)
Comment (KAndreeva) [reply src/igvfd/tests/test_schema_whole_organism.py:26]: I know. My bad!

File: src/igvfd/types/sample.py [hunk 4]
```diff
@@ -130,3 +130,15 @@ class DifferentiatedCell(Biosample):
 class TechnicalSample(Sample):
     item_type = 'technical_sample'
     schema = load_schema('igvfd:schemas/technical_sample.json')
+
+
+@collection(
+    name='whole-organisms',
+    unique_key='accession',
+    properties={
+        'title': 'Whole Organism Samples',
```
Comment (gabdank) [inline src/igvfd/types/sample.py:139]: I think the title should be "Whole organism samples" 
I don't think each word should be capitalized

**Review by gabdank — CHANGES_REQUESTED** [review]

File: src/igvfd/schemas/whole_organism.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
+{
+    "title": "Whole Organism",
```
Comment (gabdank) [inline src/igvfd/schemas/whole_organism.json:2]: I think the title should be "Whole organism", not both words capitalized

Comment (KAndreeva) [reply src/igvfd/schemas/whole_organism.json]: Done

File: src/igvfd/types/sample.py (hunk 4 above)
Comment (KAndreeva) [reply src/igvfd/types/sample.py]: Done

**Review by gabdank — APPROVED** [review]

**Review by gabdank — APPROVED** [review]

File: src/igvfd/schemas/whole_organism.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            "$ref": "mixins.json#/references"
+        },
+        {
+            "$ref": "mixins.json#/taxa"
+        }
+    ],
+    "dependentSchemas": {
+        "$merge": [
+            "biosample.json#/dependentSchemas"
```
Comment (gabdank) [inline src/igvfd/schemas/whole_organism.json:53]: I think we need to refine dependency not to include anything human

Comment (gabdank) [reply src/igvfd/schemas/whole_organism.json:53]: and add a dependency that makes sure one can not submit human taxa whole organism

**Review by gabdank — CHANGES_REQUESTED** [review]

File: src/igvfd/schemas/whole_organism.json (hunk 6 above)
Comment (KAndreeva) [reply src/igvfd/schemas/whole_organism.json:53]: Done

**Review by gabdank — CHANGES_REQUESTED** [review]
mostly comment update

File: src/igvfd/schemas/whole_organism.json [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+                                    "week"
+                                ]
+                            }
+                        }
+                    }
+                }
+            ]
+        },
+        "life_stage": {
+            "comment": "Mouse biosamples have different life stages to yeast.",
```
[the remark this answers is not in the document]
Comment (KAndreeva) [reply src/igvfd/schemas/whole_organism.json:115]: Done

File: src/igvfd/schemas/whole_organism.json [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+                        "required": [
+                            "taxa"
+                        ]
+                    },
+                    "then": {
+                        "properties": {
+                            "life_stage": {
+                                "enum": [
+                                    "adult",
+                                    "child",
```
[the remark this answers is not in the document]
Comment (KAndreeva) [reply src/igvfd/schemas/whole_organism.json:135]: deleted child

**Review by gabdank — APPROVED** [review]

**Pull request merged**